### PR TITLE
Refactor NavigationBar

### PR DESCRIPTION
**Pull request by flvndvd**
## Description

This PR advances the refactoring process by focusing on the sidebar navigation. The objective is to avoid rendering `AppLayout` on every page by adopting the [Layout Pattern](https://nextjs.org/docs/pages/building-your-application/routing/pages-and-layouts#layout-pattern) similar to what was implemented in the conversation views.

Key changes in this PR include:
- Moving all elements of the navigation bar into dedicated components.
- Removing the `topNavigationCurrent` props used for tab switching in favor of using the router.
- Ensuring the mobile navigation is completely autonomous, including the hamburger menu functionality.

## Risk

Worst case it breaks the top level navigation from the sidebar. Safe to rollback.

### CODE REVIEW

File: front/components/navigation/Navigation.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  const { sidebarOpen, setSidebarOpen } = useContext(SidebarContext);
+
+  if (hideSidebar) {
+    return null;
+  }
+
+  return (
+    <div className="flex shrink-0 overflow-x-hidden">
+      {/* Mobile sidebar */}
+      <div className="fixed left-0 top-0 z-40 flex h-16 shrink-0 items-center gap-x-4 px-4 lg:hidden lg:px-6">
```
Comment (JulesBelveze) [inline front/components/navigation/Navigation.tsx:35]: What about having a sidebar component? As we have one for the desktop sidebar

**Comment by tdraier** [comment]
In order to avoir rerendering of the layout when using next router, shouldn't we use the `Page.getLayout` function instead of calling the Layout directly in page rendering ? Page.getLayout allows to put a component around the nested router, and completely avoid re-rendering (but i think it makes more sense when having a layout at the app level, not sure how it works at page level)

**Review by tdraier — APPROVED** [review]

**Comment by flvndvd** [comment]
> In order to avoir rerendering of the layout when using next router, shouldn't we use the `Page.getLayout` function instead of calling the Layout directly in page rendering ? Page.getLayout allows to put a component around the nested router, and completely avoid re-rendering (but i think it makes more sense when having a layout at the app level, not sure how it works at page level)

Yes that's exactly the idea, we did something similar in https://github.com/dust-tt/dust/pull/4349 😉.

File: front/components/navigation/Navigation.tsx (hunk 1 above)
Comment (flvndvd) [reply front/components/navigation/Navigation.tsx:35]: You mean one component for mobile and one for desktop?

File: front/components/navigation/config.ts [hunk 2]
```diff
@@ -59,32 +59,48 @@ export type SparkleAppLayoutNavigation = {
...
+    | SubNavigationAssistantsId
+    | SubNavigationAdminId
+    | SubNavigationAppId;
+  label: string;
+  icon: React.ComponentType<{ className?: string }>;
+  href?: string;
+  hideLabel?: boolean;
+  sizing?: "hug" | "expand";
+  hasSeparator?: boolean;
+  current?: never;
```
[the remark this answers is not in the document]
Comment (flvndvd) [reply front/components/navigation/config.ts:78]: It's while phasing out the current types.

File: front/components/navigation/Navigation.tsx (hunk 1 above)
Comment (JulesBelveze) [reply front/components/navigation/Navigation.tsx:35]: Yes we have one component for desktop "NavigationSideBar" and we could create a "MobileSideBar". We would simply conditionally render one in here

Comment (JulesBelveze) [reply front/components/navigation/Navigation.tsx:35]: I can do it in a separate PR

**Review by JulesBelveze — APPROVED** [review]

File: front/components/navigation/Navigation.tsx (hunk 1 above)
Comment (flvndvd) [reply front/components/navigation/Navigation.tsx:35]: All yours base on the implementation of your hide sidebar 🙃.

**Pull request merged**